_includes/tutorials/splitting/kstreams/code/src/main/java/io/confluent/developer/SplitStream.java

```diff
@@ -99,20 +86,21 @@ public static void main(String[] args) throws Exception {
         }
 
         SplitStream ss = new SplitStream();
-        Properties envProps = ss.loadEnvProperties(args[0]);
-        Properties streamProps = ss.buildStreamsProperties(envProps);
-        Topology topology = ss.buildTopology(envProps);
+        Properties allProps = ss.loadEnvProperties(args[0]);
+        allProps.put(StreamsConfig.DEFAULT_KEY_SERDE_CLASS_CONFIG, Serdes.String().getClass());
+        allProps.put(StreamsConfig.DEFAULT_VALUE_SERDE_CLASS_CONFIG, SpecificAvroSerde.class);
 
-        ss.createTopics(envProps);
+        Topology topology = ss.buildTopology(allProps);
+        ss.createTopics(allProps);
 
-        final KafkaStreams streams = new KafkaStreams(topology, streamProps);
+        final KafkaStreams streams = new KafkaStreams(topology, allProps);
         final CountDownLatch latch = new CountDownLatch(1);
 
         // Attach shutdown handler to catch Control-C.
         Runtime.getRuntime().addShutdownHook(new Thread("streams-shutdown-hook") {
             @Override
             public void run() {
-                streams.close();
+                streams.close(Duration.ofSeconds(5));
```

Comment:
Similar question as the other KT, what does this do?

Reply:
See doc below:
<img width="1189" alt="Screen Shot 2021-05-12 at 1 12 50 PM" src="https://user-images.githubusercontent.com/12937975/118031434-c24d3880-b323-11eb-8160-521b4d694149.png">
Without it, the application may take a while to end. With it, the threads will only wait up to 5 seconds to join and end the app. Something similar was added by @bbejeck in [this PR](https://github.com/confluentinc/kafka-tutorials/pull/812/files). It seems like something we should have in all of our kstreams apps.

Reply:
What problem does this solve?  Were there some applications take a lot longer to shutdown?

Reply:
Exactly that. We use this in many places: https://github.com/confluentinc/kafka-tutorials/search?q=close%28Duration.ofSeconds%285%29%29%3B

Reply:
👍